packages/typespec-client-generator-core/src/types.ts

```diff
@@ -215,6 +216,11 @@ export function addEncodeInfo(
       innerType.encode = "bytes";
     }
   }
+  if (isSdkIntKind(innerType.kind)) {
+    if (encodeData && "encode" in innerType) {
+      innerType.encode = getEncodeHelper(context, type, innerType.kind);
```

Comment:
i believe `getEncodeHelper` is not used to add encode from typespec, but just a kind of initialization. `addEncodeInfo` is the right place to handle real encode. you may move izzy's change to that function to here.

Reply:
Moved the code from `getEncodeHelper` to `addEncodeInfo`.

Added a test case to verify `@encode` on scalar still works.

Reply:
tcgc doesn't support it shouldn't be a reason to not add a scenario. This is a valid scenario which is also the most intuitive way to use `@encode`

Reply:
Timothee, not sure I understand your comment.

I've added case of both encode on property
```
      model Test {
        @encode(string)
        value: safeint;
      }
```
and encode on scalar
```
      @encode(string)
      scalar int32EncodedAsString extends int32;

      model Test {
        value: int32EncodedAsString;
      }
```

I assume this is all the cases we had on "integer encode as string'?

Reply:
im sorry i commented on the wrong pr after reading the message in the email, forget, does this pr actually fix this?

Reply:
yeah was about that one so forget what I said https://github.com/Azure/cadl-ranch/pull/656 seems like you are fixing it